climakitae/metadata_update.py

```diff
@@ -0,0 +1,57 @@
+# Decorator to append transform call
+# details as metadata to ds/da
+
+from functools import wraps
+def transform_details(func):      
+    @wraps(func)  
+    def update_metadata(*args, **kwargs):
+        
+        """
+        Wrapper that pulls function parameters on execution of any
+        transform 'func'. 
+        Usage: place @transform_details above transform definition.
+        
+        Inputs:
+        args: function arguments. 
+        NOTE: Assumes that the xarray.Dataset or xarray.DataArray is 
+        the first argument. 
+        kwargs: function keyword arguments. 
+        
+        Output: 
+        The resultant transformed ds/da with updated metadata.
+        """
+
+        # Getting the name, args, and kwargs of the called function
+        fname = func.__name__
+        ds = args[0]
+        func_args = args[1:]
+        func_kwargs = kwargs
+        
+        # get existing ds/da attributes as a dictionary
+        # which can be updated with new information.
+        orig_attrs = ds.attrs
+
+        # flag that climakitae transform has been applied
+        transform_attrs = {'post_processed' : 'true',
+                     'post_processed_by' : 'Cal-Adapt Analytics'+
+                     'Engine v 0.1'}
```

Comment:
I think it's version 0.0.1 at the moment? Maybe there is a way to pull that information dynamically so this doesn't have to be manually updated whenever we declare a new version number.

Reply:
It is climakitae-0.0.1

print(ck.__version__)

Reply:
@odoherty is that an actual command, or a suggestion? Because it doesn't work for me

Reply:
it is a suggestion -- you will, however, be able to get this somehow from github -- the numbering of releases is something that will manually be done there. You can see that there aren't currently any releases though -- we're intending whatever we have for the soft launch to be 0.0.1

Reply:
it may not work yet because we've not done any official releases. That's something someone will manually specify on github as 0.0.1 for the soft launch -- enshrining that version of things forever as 0.0.1 -- and then we make a new release when we have an appropriate amount of updates